src/core/main.c

```diff
@@ -747,6 +747,17 @@ static void manager_set_defaults(Manager *m) {
         manager_environment_add(m, NULL, arg_default_environment);
 }
 
+static void manager_load_environment(Manager *m) {
+        _cleanup_strv_free_ char **files = NULL;
+
+        if (arg_system)
+                return;
+
+        conf_files_list_nulstr(&files, ".conf", NULL, CONF_PATHS_NULSTR("environment.d"));
+        strv_extend(&files, "/etc/environment");
```

Comment:
If I'm reading this correctly, this will make `/etc/environment` take precedence over `/etc/environment.d/a.conf`. This would be very confusing.